internal/types/document.go

```diff
@@ -216,17 +154,14 @@ func (d *Document) add(key string, value any) error {
 		return fmt.Errorf("types.Document.add: key already present: %q", key)
 	}
 
-	if !isValidKey(key) {
-		return fmt.Errorf("types.Document.add: invalid key: %q", key)
-	}
-
+	// TODO Should we still validate value if we don't validate key anymore?
```

Comment:
@AlekSi, this place is controversial to me now. I like the fact that we validated both `key` and `value` here. I think, for consistency, `isValidKey` should be called `validateKey` and should return the error, but overall the idea looked reasonable. Now, if we remove this `key` validation, we lose a part of the logic.